Serilog.Sinks.Network/Sinks/TCP/TCPSocketWriter.cs

```diff
@@ -69,34 +69,35 @@ public TcpSocketWriter(Uri uri, int maxQueueSize = 5000)
         {
             _eventQueue = new FixedSizeQueue<string>(maxQueueSize);
             _tokenSource = new CancellationTokenSource();
-            
-                Func<Uri, Stream> tryOpenSocket = h =>
+
+            Func<Uri, Task<Stream>> tryOpenSocket = async h =>
+            {
+                try
                 {
-                    try
-                    {
-                        TcpClient client = new TcpClient(uri.Host, uri.Port);
-                        Stream stream = client.GetStream();
-                        if (uri.Scheme.ToLower() != "tls")
-                            return stream;
+                    TcpClient client = new TcpClient();
+                    await client.ConnectAsync(uri.Host, uri.Port);
+                    Stream stream = client.GetStream();
+                    if (uri.Scheme.ToLower() != "tls")

```

Comment:
`if (uri.Scheme.ToLower() != "tls")` check it once, before loop, to avoid string allocations due to .ToLower()